src/vs/editor/contrib/codeAction/browser/codeActionController.ts

```diff
@@ -251,6 +252,10 @@ export class CodeActionController extends Disposable implements IEditorContribut
 			},
 			onHide: () => {
 				this._editor?.focus();
+			},
+			onFocus: async (action: CodeActionItem) => {
+				await action.resolve(CancellationToken.None);
```

Comment:
Can we cancel this when the code action widget hides, either because the user dismissed it or because they selected another action

Also just double check that hovering multiple times only results in a single `resolve` for a code action that takes a long time to resolve